src/compute-client/src/plan/transform/api.rs

```diff
@@ -9,17 +9,23 @@
 
 //! Utilities for transformation of [crate::plan::Plan] structures.
 
+use std::collections::BTreeSet;
+
 use mz_ore::stack::RecursionLimitError;
+use mz_repr::GlobalId;
 
 use crate::plan::interpret::{BoundedLattice, FoldMut, Interpreter};
 use crate::plan::Plan;
 
 /// The type of configuration options passed to all [Transform::transform] calls
 /// as an immutable reference.
-pub type TransformConfig = ();
+#[derive(Debug)]
+pub struct TransformConfig {
+    pub monotonic_ids: BTreeSet<GlobalId>,
+}
 
 /// A transform for [crate::plan::Plan] nodes.
-pub trait Transform<T = mz_repr::Timestamp> {
+pub trait Transform<'a, T = mz_repr::Timestamp> {
```

Comment:
nit: strictly speaking the `'a` bound on `config: &'a TransformConfig` is not needed any more in this PR, so maybe we can minimize the diff and punt on evolving this part of the API until we needed (and I'm sure we'll need something like that if we keep adding LIR ⇒ LIR transforms).

For example, in practice it might turn out that we need to constrain `&self` to be `&'b self` where `'b: 'a`, but I'm not sure.

Reply:
To address this, I applied @antiguru's suggested patch (thanks, both!).